### PR TITLE
MRG: Fix tests

### PR DESCRIPTION
**Pull request by Eric89GXL**
Hopefully this makes our CIs more robust and actually pass.

### CODE REVIEW

File: mne/cov.py [hunk 1]
```diff
@@ -1129,6 +1129,12 @@ def _get_ch_whitener(A, pca, ch_type, rank):
     """"Get whitener params for a set of channels."""
     # whitening operator
     eig, eigvec = linalg.eigh(A, overwrite_a=True)
+
+    # Clean up our eigenvalues (can be e.g. -1e-26 or 1e-26 for MEG data)
+    eps = np.finfo(float).eps
+    tol = np.max(A.shape) * np.amax(eig) * eps
+    eig[np.abs(eig) < tol] = 0
```
Comment (Eric89GXL) [inline mne/cov.py]: Turns out this was the underlying bug via this path:
```
mne.inverse_sparse._gamma_map.gamma_map
mne.inverse_sparse.mxne_inverse._prepare_gain
mne.inverse_sparse.mxne_inverse._prepare_gain_column
mne.minimum_norm.inverse._prepare_forward
mne.cov.prepare_noise_cov
mne.cov._get_ch_whitener
```
For all builds except 2.7 (for whatever reason), the first eigenvalue was -1e-27, so the check in `_prepare_forward` of `nzero = (eig > 0)` correctly removed the first component. But for 2.7, it was 1e26 (slightly positive) so some horrible noise was added to the whitener. Not sure how we've been lucky enough to avoid this previously. I think this is potentially related to the re-application of projectors, but I'm not sure.

This fix uses the same sort of tolerance used by `orth` to determine what the cutoff bound should be. Hopefully it's good enough...

File: mne/utils.py [hunk 2]
```diff
@@ -381,7 +382,13 @@ def estimate_rank(data, tol=1e-4, return_singular=False,
         norms = _compute_row_norms(data)
         data /= norms[:, np.newaxis]
     s = linalg.svd(data, compute_uv=False, overwrite_a=True)
-    rank = np.sum(s >= tol)
+    if isinstance(tol, string_types):
+        if tol != 'auto':
+            raise ValueError('tol must be "auto" or float')
+        eps = np.finfo(float).eps
+        tol = np.max(data.shape) * np.amax(s) * eps
```
Comment (Eric89GXL) [inline mne/utils.py]: This is the more principled fix. Turns out the bug was really with our `rank` calculations. Although our rank calculations had been wrong, when they were used later, we got lucky that the eigenvalues were small *and* negative, e.g. `-1e-26` for MEG covariance (where non-zero ones live around 1e-12), and our code used the check `nzero = (eig > 0)` in compute_whitener. But for whatever reason, on this 2.7 build they were small but positive, e.g. `1e-26` so the check failed to exclude them. Fixing the rank calculation this way -- by setting the tolerance to a reasonable value in the same way as `linalg.orth` does -- fixes the rank estimation, which then sets the `eig` value to zero explicitly in `_get_ch_whitener`. Whew!

Comment (Eric89GXL) [inline mne/utils.py]: (note that the old value of `tol=1e-4` was *close* to correct, e.g. for this particular failure the "zero" eigenvalues were 1e-3 after the normalizations -- so the errors were hopefully rare)

**Comment by Eric89GXL** [comment]
GREEN TRAVIS

Ready for review/merge from my end.

**Comment by jona-sassenhagen** [comment]
>GREEN TRAVIS

haha finally

File: mne/cov.py [hunk 3]
```diff
@@ -1808,7 +1808,7 @@ def _check_scaling_inputs(data, picks_list, scalings):
     return scalings_
 
 
-def _estimate_rank_meeg_signals(data, info, scalings, tol=1e-4,
+def _estimate_rank_meeg_signals(data, info, scalings, tol='auto',
```
Comment (agramfort) [inline mne/cov.py]: @dengemann please check this change

File: mne/utils.py (hunk 2 above)
Comment (dengemann) [inline mne/utils.py]: But we changed the old values because they did not regularize enough. Should we check our PRNI paper script or the inverse cov examples before merging? Did you compare how this option behaves in critical situations?

Comment (dengemann) [inline mne/utils.py]: Sorry Eric, I got confused. These changes are unrelated to the modifications described in our PRNI paper.

**Comment by agramfort** [comment]
LGTM

+1 for merge if @dengemann is ok.

**Comment by dengemann** [comment]
yes

**Comment by agramfort** [comment]
thx @Eric89GXL !
